dht_net.go

```diff
@@ -109,40 +108,62 @@ func (dht *IpfsDHT) handleNewMessage(s inet.Stream) bool {
 
 // sendRequest sends out a request, but also makes sure to
 // measure the RTT for latency measurements.
-func (dht *IpfsDHT) sendRequest(ctx context.Context, p peer.ID, pmes *pb.Message) (*pb.Message, error) {
-
-	ms, err := dht.messageSenderForPeer(ctx, p)
+func (dht *IpfsDHT) sendRequest(ctx context.Context, p peer.ID, req *pb.Message) (*pb.Message, error) {
+	ps, err := dht.getStream(ctx, p)
 	if err != nil {
 		return nil, err
 	}
-
 	start := time.Now()
-
-	rpmes, err := ms.SendRequest(ctx, pmes)
+	replyChan, err := ps.request(ctx, req)
 	if err != nil {
+		ps.reset()
 		return nil, err
 	}
-
-	// update the peer (on valid msgs only)
-	dht.updateFromMessage(ctx, p, rpmes)
-
-	dht.peerstore.RecordLatency(p, time.Since(start))
-	logger.Event(ctx, "dhtReceivedMessage", dht.self, p, rpmes)
-	return rpmes, nil
+	onReply := func(reply *pb.Message) {
+		dht.streamPool.put(ps, p)
```

Comment:
A comment for clarity:

```suggestion
		// if the stream is errored, the pool will drop it.
		dht.streamPool.put(ps, p)
```

Reply:
This has changed, see https://github.com/libp2p/go-libp2p-kad-dht/pull/285/files#diff-a4376ba51cc33635f6247068e8253508R172.